platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintChecker.kt

```diff
@@ -310,17 +314,43 @@ class ConfigurationCacheFingerprintChecker(private val host: Host) {
         }
     }
 
-    private
-    fun hasFileChanged(file: File, originalHash: HashCode) =
-        !isUpToDate(file, originalHash)
-
     private
     fun hasDirectoryChanged(file: File, originalHash: HashCode?) =
         host.hashCodeOfDirectoryContent(file) != originalHash
 
     private
-    fun isUpToDate(file: File, originalHash: HashCode) =
-        host.hashCodeOf(file) == originalHash
+    fun isFileUpToDate(file: File, originalHash: HashCode) =
+        checkFileUpToDateStatus(file, originalHash) == UpToDateStatus.Unchanged
+
+    enum class UpToDateStatus {
```

Comment:
💅 Could be private?
💅 IMO name `FileUpToDateStatus` is more precise here. And than `FileChanged` would become `ContentChanged` in symmetry with `TypeChanged`